### PR TITLE
Fix execution button delay

### LINKED ISSUE #1558: Proposal Page - Time Misalignment between Toaster execution alert and the execution button
See video at 5.23 and 3.23

https://www.loom.com/share/3af51acc15ae4f3282c3287eebcc44d8

When the toaster pops up showing a 'executed successfully' for a brief second the button on the proposal page (currently greyed out as it should be) will light up orange again. If the user clicks it, it will error (as expected) ... but why is it lighting up at all (i.e. prompting a user action)

There's clearly a lag when the UI hasn't fully caught up with the toaster on the current proposal status. 

This lag maybe impossible to fix, ...but lets explore removing the 'light up' of the button. The button should remain grey until an actual transaction possibility is on the cards

### PR DESCRIPTION
**Pull request by DarksightKellar**
Closes #1558

This PR patches up the time delay between when the app indicates a proposal timelock and/or execute transaction has been successfully ran and when the data actually indicates as much.

We have achieved this by simply keeping the execute button disabled after the success toast. If the user chooses to stay on the screen, eventually the node data will be updated, and the button will disappear.

In the case of timelock, the button will re-enable itself after timelock when the proposal becomes executable.

I expect, if the user refreshes the page in that small window where the node hasn't reflected the updated proposal state, the button will then be clickable -- not much we can do in that case beyond localstorage caching, which we _have_ discussed. Problem is multisig safe proposals are currently not cached as they're loaded all at once every time, and choosing to selectively cache them will take a bit more thought. 

There's a separate task for this caching [here](https://github.com/orgs/decentdao/projects/32/views/3?pane=issue&itemId=65769569), if we end up deciding it's something worth doing.

### CODE REVIEW

**Comment by adamgall** [comment]
My take is that trying to handle that small timeframe window of a user potentially refreshing the page, using some local storage caching, is way overkill.

File: src/components/Proposals/MultisigProposalDetails/TxActions.tsx [hunk 1]
```diff
@@ -30,12 +31,25 @@ export function TxActions({ proposal }: { proposal: MultisigProposal }) {
   const signerOrProvider = useSignerOrProvider();
   const safeAPI = useSafeAPI();
 
+  const [isSubmitDisabled, setIsSubmitDisabled] = useState(false);
+
+  let wasTimelocked = useRef(
```
Comment (adamgall) [inline src/components/Proposals/MultisigProposalDetails/TxActions.tsx:36]: `const`, no?

File: src/components/Proposals/MultisigProposalDetails/TxActions.tsx [hunk 2]
```diff
@@ -181,6 +199,7 @@ export function TxActions({ proposal }: { proposal: MultisigProposal }) {
         pendingMessage: t('pendingExecute', { ns: 'transaction' }),
         successMessage: t('successExecute', { ns: 'transaction' }),
         successCallback: async () => {
+          setIsSubmitDisabled(true);
```
Comment (adamgall) [inline src/components/Proposals/MultisigProposalDetails/TxActions.tsx:202]: Do we maybe want to set this property when the user initially clicks the "execute" button? or does it need to be in this `successCallback` for a specific reason?

File: src/components/Proposals/MultisigProposalDetails/TxActions.tsx [hunk 3]
```diff
@@ -125,8 +139,9 @@ export function TxActions({ proposal }: { proposal: MultisigProposal }) {
         failedMessage: t('failedExecute', { ns: 'transaction' }),
         pendingMessage: t('pendingExecute', { ns: 'transaction' }),
         successMessage: t('successExecute', { ns: 'transaction' }),
-        successCallback: async () => {
-          await loadSafeMultisigProposals();
+        successCallback: () => {
+          setIsSubmitDisabled(true);
+          loadSafeMultisigProposals();
```
Comment (adamgall) [inline src/components/Proposals/MultisigProposalDetails/TxActions.tsx:144]: Doesn't this call to `loadSafeMultisigProposals` need to be `async`?

File: src/hooks/DAO/loaders/useLoadDAOProposals.ts [hunk 4]
```diff
@@ -15,20 +15,21 @@ export const useLoadDAOProposals = () => {
...
+      return loadAzoriusProposals(proposal => {
         action.dispatch({
           type: FractalGovernanceAction.SET_AZORIUS_PROPOSAL,
           payload: proposal,
         });
       });
     } else if (type === GovernanceType.MULTISIG) {
       // load mulisig proposals
+      // @dev what is the point of setMethodOnInterval here?
       setMethodOnInterval(loadSafeMultisigProposals);
```
Comment (adamgall) [inline src/hooks/DAO/loaders/useLoadDAOProposals.ts:33]: Doesn't this call to `loadSafeMultisigProposals` need to be `async`?

**Review by adamgall — CHANGES_REQUESTED** [review]
@DarksightKellar can you please follow this up with a PR to address the various `@todo`s you created for the names of various types/interfaces/properties?

File: src/components/Proposals/MultisigProposalDetails/TxActions.tsx (hunk 1 above)
Comment (DarksightKellar) [reply src/components/Proposals/MultisigProposalDetails/TxActions.tsx:36]: `const`, yep

File: src/components/Proposals/MultisigProposalDetails/TxActions.tsx (hunk 2 above)
Comment (DarksightKellar) [reply src/components/Proposals/MultisigProposalDetails/TxActions.tsx:202]: This was to account for potential errors (after which the button would need to be "re-enabled"). This way only successful executes would actually disable the button.

File: src/hooks/DAO/loaders/useLoadDAOProposals.ts (hunk 4 above)
Comment (DarksightKellar) [reply src/hooks/DAO/loaders/useLoadDAOProposals.ts:33]: Eh I didn't think so at first, but I suppose in `setMethodOnInterval`'s first-time ran line `returnValue = await Promise.resolve(getMethod());`, awaiting the call will actually wait for `loadSafeMultisigProposals` to complete?

**Comment by tomstuart123** [comment]
Nice @DarksightKellar / @adamgall .

I tested this with the same childSafe [here ](https://deploy-preview-1990.app.dev.decentdao.org/home?dao=sep:0x773eC5B85aebAa323e700E9bbAA7Cb2eFeAdf56F)where I was originally having issues. The timelock and execute button remained greyed out during the process after click/tx.


Approved

**Review by tomstuart123 — APPROVED** [review]

**Comment by adamgall** [comment]
@DarksightKellar i fixed the merge conflict and pushed, fyi

**Review by adamgall — APPROVED** [review]

**Comment by adamgall** [comment]
@DarksightKellar does this PR address #1511 also?

**Comment by Da-Colon** [comment]
> @DarksightKellar does this PR address #1511 also?

It helps it for sure but doesn't 'fix' it. With this fix relies on the user not refreshing, right? The button would reappear at that point able to be pushed?

**Comment by DarksightKellar** [comment]
> > @DarksightKellar does this PR address #1511 also?
> 
> It helps it for sure but doesn't 'fix' it. With this fix relies on the user not refreshing, right? The button would reappear at that point able to be pushed?

@Da-Colon I had the same observation. See my comment on this: 
> I expect, if the user refreshes the page in that small window where the node hasn't reflected the updated proposal state, the button will then be clickable -- not much we can do in that case beyond localstorage caching, which we have discussed. Problem is multisig safe proposals are currently not cached as they're loaded all at once every time, and choosing to selectively cache them will take a bit more thought.
> There's a separate task for this caching [here](https://github.com/orgs/decentdao/projects/32/views/3?pane=issue&itemId=65769569), if we end up deciding it's something worth doing.

And @adamgall's take:
> My take is that trying to handle that small timeframe window of a user potentially refreshing the page, using some local storage caching, is way overkill, and not something we need to try to solve for.

which I'm inclined to agree with. It's such a uniquely specific, yet non-breaking flow, that solving for it specifically using caching feels overkill. If we agree that caching multisig safe proposals _is_ worth doing though (I can't argue for it atp), then this will sort itself out as a side effect.

**Comment by Da-Colon** [comment]
> @Da-Colon @DarksightKellar so is #1511 related to the button re-appearing only after immediately refreshing after it was executed the first time?
> 
> If so, I'd say we can close #1511 as a wontfix

Coolio I agree with that

**Pull request merged**